File: 2 Pointers/Typescript/sortedSquares.ts
function sortedSquares(nums: number[]): number[] {
    let high = nums.length - 1
    let low = 0
    const result: number[] = [];

    while (low <= high) {
        if (nums[low] ** 2 > nums[high] ** 2){
            result.push(nums[low]**2)
            low += 1
        }
        else{
            result.push(nums[high]**2)
            high -= 1
        }
    }
    result.reverse()
  return result;
};

File: 2 Pointers/Python/sortedSquares.py
def sortedSquares(self, nums: List[int]) -> List[int]:
        low = 0
        high = len(nums) - 1
        result = []
        while low <= high:
            if abs(nums[low]) > abs(nums[high]):
                result.append(nums[low]**2)
                low += 1
            else:
                result.append(nums[high]**2)
                high -= 1
        result.reverse()
        return result

'''
Time complexity: O(n)
Space complexity: O(1)
'''
